### ISSUE TITLE
Release skipped on CircleCI?

### ISSUE THREAD

**Issue by simlu** [semantic-release/semantic-release#1074]
Unfortunately I'm having trouble publishing. My setup is as follows:

(1) Only one master branch on main repo and a forked repo. PRs should be merged from fork into master and then directly published.
(2) The CircleCI option "Build forked pull requests" is set to "off"
(3) The CircleCI build are now triggered once the PR is merged in. That part works great. However the environment variables [CIRCLE_PULL_REQUEST, CIRCLE_PR_NUMBER, etc](https://circleci.com/docs/2.0/env-vars/) are still set.

This now causes semantic-release to stall with the error message `This run was triggered by a pull request and therefore a new version won't be published.`

Any idea how I could fix this?

Thank you very much! Cheers, L~

PS: I'm pretty sure on TravisCI this behaviour is different.

**Comment by pvdlg** [comment]
What does trigger the CircleCI build, pushing to the PR branch or merging to master? On which branch CircleCI is running?

If the job was triggered by the merge to master and the build actually run on master branch then `CIRCLE_PULL_REQUEST` and `CIRCLE_PR_NUMBER` should not be set. If they are that's an issue with CircleCI that you should report over there.

**Comment by simlu** [comment]
The build is triggered by merging into master. I'll try circle ci and see what they have to say and get back here. Cheers!

**Comment by simlu** [comment]
Got a response when describing the above behavior to them:

*[...] The merge is part of the PR so these variables are expected to be set. The `CIRCLE_PULL_REQUEST` is specific to a forked PR and `CIRCLE_PR_NUMBER` references the specific PR or one chosen at random if there are multiples associated. More on this can be found in the [built-in environment](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables) variables documentation.*

So these are expected to be set. Do we need a flag `--allow-on-pr`? How should this best be solved?

Cheers, L~

**Comment by pvdlg** [comment]
Last time I tried CircleCI those variable were set only on the merge build triggered by opening/updating the PR. After merging the PR in GitHub a regular build was triggered on master.

Can you provide a link to the problematic CircleCI job?

**Comment by simlu** [comment]
Unfortunately the repo is private atm. I can probably set up a test case. But this might have to do with (2)?

**Comment by simlu** [comment]
@pvdlg I've put a pr up for now: https://github.com/semantic-release/semantic-release/pull/1076

As mentioned, this solves my use case nicely. If you prefer to see a real world example of why this is necessary I might be able to set something up. If you are currently using circle it might be very easy for you to try it out though (the setup is very simple).

**Comment by pvdlg** [comment]
I just tried to reproduce that on CircleCI and everything works as expected.
I opened the PR https://github.com/pvdlg/playground/pull/38 which triggered the build https://circleci.com/gh/pvdlg/playground/57 where `CIRCLE_PULL_REQUEST` is set as expected.

Then I merged the PR to `master` which triggered the build https://circleci.com/gh/pvdlg/playground/59 in which `CIRCLE_PULL_REQUEST` is not as expected.

So as far as I can tell CircleCI works as expected, and semantic-release will make a release during the build triggered on `master`.

**Comment by simlu** [comment]
@pvdlg It appears that you didn't use a fork, which is crucial.

**Comment by pvdlg** [comment]
If you do that the PR will not be built, only the merge to master will.

Tested again with "Build forked pull requests" set to "off" with PR https://github.com/pvdlg/playground/pull/40.

Same result, the merge triggers the job https://circleci.com/gh/pvdlg/playground/67 on `master` and `CIRCLE_PULL_REQUEST` is not set.

**Comment by simlu** [comment]
@pvdlg Yes, that part is expected (does not build when you open the PR). But I'm really curious as to why the PR variables are not set for you... This is certainly not what I am seeing. Maybe it has something to do with private repos?

I will do some digging tomorrow (it's really late here) and get back with some details!

**Comment by simlu** [comment]
And thank you very much for trying to reproduce this. I swear this is *not* a user (me) error. I'll proof it tomorrow.

**Comment by simlu** [comment]
For reference, these are my settings.

![settings](https://user-images.githubusercontent.com/1539747/51424552-45ab5480-1b84-11e9-923e-4297160a6dc3.png)

**Comment by simlu** [comment]
I tried to reproduce the behaviour with a separate public / private repo and tried a bunch of different things. I was not able to reproduce the behaviour that I'm seeing on my repo. 

However I can reliably reproduce the behaviour on my existing repo. Hence I'm considering this a CircleCi bug again. I will go back at their support.... #timesink #ffs

**Comment by simlu** [comment]
Their support got back to me:

> We do have one known issue where the pull request variable can be set inadvertently. This happens when someone submits a pull request from a fork in this manner:
> 
> - Fork the project
> - Wait for a change to the upstream to be made
> - Create a pull request on the fork requesting to merge the upstream's default branch into the fork
> 
> Future builds on the upstream project will have the CIRCLE_PULL_REQUEST variable set to the fork's new PR details. Is it possible someone could have created a pull request in this way on that project?

This doesn't appear to be my exact use case, but it indicates that they're not confident they figured all the cases out. The flow described doesn't seem to be that uncommon and is similar to what I was doing.

I'm closing this issue. Hopefully it'll save someone else some time :(

**Comment by simlu** [comment]
I've finally figured out what is going on and can reproduce. The problem was / is that my circleci workflow creates a new pr in the first job. This then messes up subsequent jobs.

I'll update this issue when I get more information from CirlceCI

**Comment by cgadam** [comment]
@simlu I am having the same exact problem than the one you mentioned with a specific private repo. It seems those variables are being populated even for the merge commit coming into master. Did you figure this out finally? Was the workflow creating a PR? In my case I'm pretty sure we are not creating any PR automagically. Thanks!

**Comment by simlu** [comment]
@cgadam Yeah that was the problem. So if you are not creating any PRs in your workflow your cause is likely elsewhere.

**Comment by Doginal** [comment]
For people coming from google, I had this issue crop up yesterday and it turned out there was a PR open from my base branch to another which caused circleci to always add that PR number to my builds. Closing that PR and everything fell into place, check your PR's!

**Comment by danguilherme** [comment]
I just had this problem with Github Actions.
Sorry for reviving this issue, let me know if I need to open a new one.

In my Action configuration, you will find this:

[.github/workflows/release.yml](https://github.com/danguilherme/uno/blob/61865ae5b97428dea39c541df53df5a4127acf14/.github/workflows/release.yml#L3-L7)
```yaml
name: Release

on:
  pull_request:
    types: [closed]
    branches:
      - master

...
```

I configured `semantic-release` in this PR: https://github.com/danguilherme/uno/pull/32.

After merging the PR to master, the [`Release` action ran](https://github.com/danguilherme/uno/runs/574862858), but `semantic-release` did not perform the release:

```
[6:37:29 PM] [semantic-release] › ℹ  Running semantic-release version 17.0.0
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/git"
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[6:37:29 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[6:37:29 PM] [semantic-release] › ℹ  This run was triggered by a pull request and therefore a new version won't be published.
```

Is this the expected behaviour?

**Comment by Ridermansb** [comment]
hi @danguilherme did you solved the problem?

I also have this issue.

**Comment by danguilherme** [comment]
@Ridermansb unfortunately not... Still triggering the releases manually when needed.

**Comment by shelooks16** [comment]
@Ridermansb, @danguilherme

Hi, I had exactly the same setup as **danguilherme** provided.

I resolved this by setting ci to false in semantic-release config file:
```
// package.json

"release": {
  "branches": [
    "master"
  ],
  "dryRun": false,
  "ci": false,
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    [
      "@semantic-release/github",
      {
        "successComment": false,
        "failComment": false
      }
    ],
    [
      "@semantic-release/changelog",
      {
        "changelogFile": "CHANGELOG.md"
      }
    ],
    [
      "@semantic-release/git",
      {
        "assets": [
          "CHANGELOG.md"
        ]
      }
    ]
  ]
}
```
